src/extra/fromEvent.ts

```diff
@@ -11,29 +13,49 @@ export class DOMEventProducer implements InternalProducer<Event> {
 
   _start(out: InternalListener<Event>) {
     this.listener = (e) => out._n(e);
-    const {node, eventType, useCapture} = this;
-    node.addEventListener(eventType, this.listener, useCapture);
+    this.node.addEventListener(this.eventType, this.listener, this.useCapture);
   }
 
   _stop() {
-    const {node, eventType, listener, useCapture} = this;
-    node.removeEventListener(eventType, listener, useCapture);
+    this.node.removeEventListener(this.eventType, this.listener, this.useCapture);
     this.listener = null;
   }
 }
 
+export class NodeEventProducer implements InternalProducer<any> {
+  public type = 'fromEvent';
+  private listener: Function;
+
+  constructor(private node: EventEmitter, private eventName: string) { }
+
+  _start(out: InternalListener<any>) {
+    this.listener = (e: any) => out._n(e);
+    this.node.addListener(this.eventName, this.listener);
+  }
+
+  _stop() {
+    this.node.removeListener(this.eventName, this.listener);
+    this.listener = null;
+  }
+}
+
+function isEmitter(element: any): boolean {
+    return element.addListener;
```

Comment:
`addListener` and also `emit`, just because xstream Streams also have `addListener`. :)